tests/python/test_platform.py

```diff
@@ -13,3 +15,32 @@ def test_preinstantiated_platforms():
     assert isinstance(vmq_1ha_car(), Platform)
     assert isinstance(simple_linearpath(), Platform)
     assert isinstance(tripod(), Platform)
+
+
+def test_platform_settings_mls():
+    survey = Survey.from_xml("data/surveys/demo/mls_wheat_demo.xml")
+
+    platform_settings = PlatformSettings(x=10, y = 0,)
+    scanner_settings = ScannerSettings(pulse_frequency=1000)
+    platform_settings.force_on_ground(survey.scene)
+    survey.add_leg(
+        platform_settings=platform_settings,
+        scanner_settings=scanner_settings,
+    )
+
+    assert math.isclose(platform_settings.z, survey.legs[0].platform_settings._cpp_object.position[2])
+
+def test_platform_settings_tls():
+    survey = Survey.from_xml("data/surveys/demo/tls_arbaro_demo_angular_resolution.xml")
+
+    platform_settings = PlatformSettings(x=10, y = 0,)
+
+    platform_settings.force_on_ground(survey.scene)
+    survey.add_leg(
+        platform_settings=platform_settings,
+    )
+
+    assert math.isclose(platform_settings.z, survey.legs[0].platform_settings._cpp_object.position[2])
```

Comment:
Does this really test that the forcing on ground was correctly executed? I think it rather checks whether platform settings have been correctly propagated to the leg (which is assume is always true). I would expect a check against whatever the groundplane of that loaded demo scene is.